Update: Port for Sphinx 8.x.x, fix path naming and Cargo.toml compatibility

nicu1989 opened the pull request:
Implements/fixes:

- [x] Works with Sphinx 8.x+ (latest)

- [x] Fixes working with modern Cargo project layouts (support workspace versioning and TOML)
https://github.com/eclipse-score/score/issues/1522

- [x] Fixes docutils Path/StrPath objects
https://github.com/eclipse-score/score/issues/1487
https://github.com/eclipse-score/score/issues/1488

nicu1989 committed:
Port: Ensure build compatibility with Sphinx 8.x.x

Tested both with sphinx 7.4.7 and sphinx 8.1.3.

Signed-off-by: Nicolae Dicu <nicolae.dicu.ext@qorix.ai>

nicu1989 committed:
Port: Added requirements.in update

Allows integration via pip.

Signed-off-by: Nicolae Dicu <nicolae.dicu.ext@qorix.ai>

nicu1989 committed:
Fix: Fixed handling of rust path

Resolved to folders instead of creating link with "::".

Resolves:

https://github.com/eclipse-score/score/issues/1487
and
https://github.com/eclipse-score/score/issues/1488

Signed-off-by: Nicolae Dicu <nicolae.dicu.ext@qorix.ai>

nicu1989 committed:
Fix: Use cargo_metadata for Cargo.toml parsing

The module used old implementation of Cargo.toml parsing.

Resolves: https://github.com/eclipse-score/score/issues/1522

Signed-off-by: Nicolae Dicu <nicolae.dicu.ext@qorix.ai>

chrisjsewell commented:
Cheers! will have a look soon

chrisjsewell committed:
Update .readthedocs.yaml

chrisjsewell reviewed (changes requested):
I haven't fully looked through the PR just yet, but if you could firstly fix the pre-commit issues that would be great thanks (see https://github.com/sphinx-extensions2/sphinx-rust/blob/main/CONTRIBUTING.md)

chrisjsewell commented on pyproject.toml:
```diff
@@ -37,7 +37,7 @@ classifiers = [
 ]
 dependencies = [
-  "sphinx~=7.3"
+  "sphinx>=7.3"
```
```suggestion
  "sphinx>=8,<9"
```

just support sphinx 8, otherwise the test infrastructure should be updated to test both 7 and 8

nicu1989 replied:
done

chrisjsewell committed:
Update .readthedocs.yaml

fix: error: package `cargo-platform v0.1.9` cannot be built because it requires rustc 1.78 or newer, while the currently active rustc version is 1.75.0

nicu1989 committed:
Update after review

Signed-off-by: Nicolae Dicu <nicolae.dicu.ext@qorix.ai>

ubmarco commented:
@nicu1989 can you check for the failing CI?

nicu1989 committed:
Fix: Drop python 3.9 support

Sphinx 8.0 dropped python 3.9 support.

Signed-off-by: Nicolae Dicu <nicolae.dicu.ext@qorix.ai>

nicu1989 commented:
@ubmarco I have fixed the CI. Dropped support for python 3.9 according to sphinx 8.0 docs: https://www.sphinx-doc.org/en/master/changes/8.0.html 

Can a maintainer trigger the check CI please?
